### PR TITLE
device: Added detection for Cisco SPA devices

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

Fixes: #8128

### CODE REVIEW

**Comment by tracernz** [comment]
Neat. I will test this if you like (I have an SPA112 at home). :+1:

**Comment by tracernz** [comment]
I don't think the system oid is correct. The one you have used doesn't exist:
```
$ snmpwalk -v2c -cpublic ata.bertie.nz 1.3.6.1.4.1.9.6.1.23.1.1
iso.3.6.1.4.1.9.6.1.23.1.1 = No Such Object available on this agent at this OID
$ snmpwalk -v2c -cpublic ata.bertie.nz | grep SPA   
iso.3.6.1.2.1.25.1.4.0 = STRING: "console=ttyS1,115200n8 rootfstype=squashfs noalign half_image=0 verify=y Hw_Model=SPA112 Router_Mode=0
iso.3.6.1.2.1.47.1.1.1.1.2.1 = STRING: "SPA112, 2 FXS"
iso.3.6.1.2.1.47.1.1.1.1.7.1 = STRING: "SPA112"
iso.3.6.1.2.1.47.1.1.1.1.13.1 = STRING: "SPA112"
```
Discovery: https://p.libren.ms/view/de12d1e9
Poller: https://p.libren.ms/view/ec796a52
Bulk walk: https://p.libren.ms/view/9512ae68

I might have some time to look at this more on Friday.

**Comment by laf** [comment]
@tracernz That's not how OIDs work. Look in your snmpwalk output and you will see the OID is returned from sysObjectID

**Comment by tracernz** [comment]
My bad. I always assumed it was a pointer.

The sysDescr regex doesn't look right (mine returns "Linux ata 2.6.26.5 #1 PREEMPT Wed Aug 2 14:04:45 CST 2017 armv5tejlI"). I wonder if @thetravellor's hostname was set to SPA112? Mine is "ata".

**Comment by tracernz** [comment]
Confirmed that the sysDescr string is `Linux <hostname> 2.6.26.5 #1 PREEMPT Wed Aug 2 14:04:45 CST 2017 armv5tejl` (hostname is configured in Network Setup > Basic Setup > Internet Settings).

File: includes/definitions/ciscospa.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+over:
+    - { graph: device_bits, text: 'Device Traffic' }
+    - { graph: device_processor, text: 'CPU Usage' }
+    - { graph: device_mempool, text: 'Memory Usage' }
+discovery:
+    -
+        sysObjectID:
+            - .1.3.6.1.4.1.9.6.1.23.1.1.1
+        sysDescr_regex:
+            - '/^(CISCO|Linux) SPA/'
```
Comment (murrant) [inline includes/definitions/ciscospa.yaml]: Why check the sysDescr?  It seems as thought that can vary quite a bit.

**Comment by tracernz** [comment]
Works great. :+1:

**Pull request merged**